Core/ComponentBaseClasses/elxTransformBase.hxx

```diff
@@ -388,16 +388,36 @@ TransformBase<TElastix>::ReadFromFile()
        * is not the same as this transform parameter file. Otherwise,
        * we will have an infinite loop.
        */
-      std::string fullFileName1 = itksys::SystemTools::CollapseFullPath(fileName);
-      std::string fullFileName2 = itksys::SystemTools::CollapseFullPath(configuration.GetParameterFileName());
-      if (fullFileName1 == fullFileName2)
+
+      const std::string configurationParameterFileName = configuration.GetParameterFileName();
+
+      if (itksys::SystemTools::CollapseFullPath(fileName) ==
+          itksys::SystemTools::CollapseFullPath(configurationParameterFileName))
       {
         itkExceptionMacro("ERROR: The InitialTransformParameterFileName is identical to the current "
                           "TransformParameters filename! An infinite loop is not allowed.");
       }
 
-      /** We can safely read the initial transform. */
-      this->ReadInitialTransformFromFile(fileName.c_str());
+      const auto lastConfigurationParameterFilePathSeparator = configurationParameterFileName.find_last_of("\\/");
+      const char firstFileNameLetter = fileName.front();
+      const bool startsWithFilePathSeparator = firstFileNameLetter == '\\' || firstFileNameLetter == '/';
+      const bool hasDriveLetter = !startsWithFilePathSeparator && firstFileNameLetter > 0 &&
+                                  std::isalpha(firstFileNameLetter) && fileName.size() > 1 && fileName[1] == ':';
+
+      if (lastConfigurationParameterFilePathSeparator != std::string::npos && !startsWithFilePathSeparator &&
+          !hasDriveLetter && !itksys::SystemTools::FileExists(fileName))
+      {
```

Comment:
I could reverse `if` and `else`, and then do:

    if (lastConfigurationParameterFilePathSeparator == std::string::npos 
        || startsWithFilePathSeparator
        || hasDriveLetter
        || itksys::SystemTools::FileExists(fileName))

To make the condition _slightly_ simpler, maybe.

Could even just test on `itksys::SystemTools::FileExists(fileName)` and forget about the other three conditions. Might make the code slightly slower sometimes, but slightly easier to read.